apps/balancer-tools/src/components/ui/date-range-picker.tsx

```diff
@@ -0,0 +1,549 @@
+                    </div>
+                  )}
+                </div>
+              </div>
+              {isSmallScreen && (
+                <Select
+                  defaultValue={selectedPreset}
+                  onValueChange={(value) => {
+                    setPreset(value);
+                  }}
+                >
+                  <SelectTrigger className="w-[180px] mx-auto mb-2">
+                    <SelectValue placeholder="Select..." />
+                  </SelectTrigger>
+                  <SelectContent>
+                    {PRESETS.map((preset) => (
+                      <SelectItem key={preset.name} value={preset.name}>
+                        {preset.label}
+                      </SelectItem>
+                    ))}
+                  </SelectContent>
+                </Select>
+              )}
+              <div>
+                <Calendar
+                  mode="range"
+                  onSelect={(value: { from?: Date; to?: Date } | undefined) => {
+                    if (value?.from != null) {
+                      setRange({ from: value.from, to: value?.to });
+                    }
+                  }}
+                  selected={range}
+                  numberOfMonths={isSmallScreen ? 1 : 2}
+                  defaultMonth={
+                    new Date(
+                      new Date().setMonth(
+                        new Date().getMonth() - (isSmallScreen ? 0 : 1),
+                      ),
+                    )
+                  }
+                />
+              </div>
+            </div>
+          </div>
+          {!isSmallScreen && (
+            <div className="flex flex-col items-end gap-1 pr-2 pl-6 pb-6">
+              <div className="flex w-full flex-col items-end gap-1 pr-2 pl-6 pb-6">
+                {PRESETS.map((preset) => (
+                  <PresetButton
+                    key={preset.name}
+                    preset={preset.name}
+                    label={preset.label}
+                    isSelected={selectedPreset === preset.name}
+                  />
+                ))}
+              </div>
+            </div>
+          )}
+        </div>
+        <div className="flex justify-end gap-2 py-2 pr-4">
+          <Button
+            onClick={() => {
+              setIsOpen(false);
+              resetValues();
+            }}
+            variant="ghost"
+          >
+            Cancel
+          </Button>
+          <Button
+            onClick={() => {
+              setIsOpen(false);
+              if (
+                !areRangesEqual(range, openedRangeRef.current) ||
+                !areRangesEqual(rangeCompare, openedRangeCompareRef.current)
+              ) {
+                onUpdate?.({ range, rangeCompare });
+              }
+            }}
+          >
+            Update
+          </Button>
+        </div>
+      </PopoverContent>
+    </Popover>
+  );
+};
+
+DateRangePicker.displayName = "DateRangePicker";
+DateRangePicker.filePath =
```

Comment:
👀

Reply:
think this is also generated by shadcn when we run `npx shadcn-ui@latest add date-range-picker`

Reply:
ok, but why keep it here? Wonder if this is a bug on their side. It doesn't look like it's the sort of thing that should be here.